blog/source/_posts/MIDI-velocity-transformer.md

```diff
@@ -0,0 +1,266 @@
+the length of the composition.
+
+We extracted 128-note samples which we quantized using 3 bins for dstart, 3 for duration and 3 for velocity.
+Pitch information remained the same.
+
+Here are **bin edges** we used:
+```
+dstart:
+  - 0.0
+  - 0.048177
+  - 0.5
+duration:
+  - 0.0
+  - 0.145833
+  - 0.450000
+velocity:
+  - 0.0
+  - 57.0
+  - 74.0
+  - 128.0
+```
+#### Quantization Samples
+<div style="-webkit-column-count: 2; -moz-column-count: 2; column-count: 2;">
+    
+  **Original** 
+        
+  **Quantized**
+
+</div>
+
+Franz Schubert / Franz Liszt: *Song Transcriptions: Aufenthalt, Gretchen am Spinnrade, Standchen von Shakespeare, Der Erlkonig*
+
+<div style="-webkit-column-count: 2; -moz-column-count: 2; column-count: 2;">
+
+  {% algrtmImg MIDI-velocity-transformer/samples/5002-real.png pianoroll 170px %}
+  {% algrtmAudio MIDI-velocity-transformer/samples/5002-real.mp3 %}
+        
+  {% algrtmImg MIDI-velocity-transformer/samples/5002-quantized.png pianoroll 170px %}
+  {% algrtmAudio MIDI-velocity-transformer/samples/5002-quantized.mp3 %}
+
+</div>
+
+
+
+Ludwig van Beethoven:  *Sonata No. 8 in C Minor, Op.13*
+<div style="-webkit-column-count: 2; -moz-column-count: 2; column-count: 2;">
+
+  {% algrtmImg MIDI-velocity-transformer/samples/9127-real.png pianoroll 170px %}
+  {% algrtmAudio MIDI-velocity-transformer/samples/9127-real.mp3 %}
+        
+  {% algrtmImg MIDI-velocity-transformer/samples/9127-quantized.png pianoroll 170px %}
+  {% algrtmAudio MIDI-velocity-transformer/samples/9127-quantized.mp3 %}
+
+</div>
+
+### Model Architecture
+{% algrtmImgBanner MIDI-velocity-transformer/transformer.png transformer%}
+A transformer built as described in [Attention is all you need](https://arxiv.org/abs/1706.03762) paper was used for this task.
+The important hyperparameters:
+| hyperparameter | number |
+| -------------- | :-----: |
+| Number of layers in encoder and decoder | **6** |
+| Nuber of heads in attention layers | **8** |
+| Dimension of encoder and decoder outputs | **512** |
+| Dimension of a hidden layer of position-wise fast-forward network from each layer of encoder and decoder | **2048** |
+
+
+### Training and Evaluation
+#### Data
+The vocabulary for our model was generated by considering every combination of quantized information from MIDI notes. The source vocabulary consisted of 795 unique values, while the target vocabulary included 131 values, accounting for start, end, and padding tokens.
+
+Our training dataset comprised approximately 200 hours of musical data sourced from the 
+[roszcz/maestro-v1](https://huggingface.co/datasets/roszcz/maestro-v1) dataset, which includes 1276 pieces of classical music performed during piano competitions. Each musical piece was segmented into 128-note sequences, with a 64-note overlap between adjacent samples. These sequences were quantized, and each note was mapped to its corresponding index in the source and target vocabularies.
```

Comment:
I'm not the owner of maestro, you should cite the original source: https://magenta.tensorflow.org/datasets/maestro